src/main/java/ekud/command/Parser.java

```diff
@@ -0,0 +1,157 @@
+        Component[] components = new Component[componentStrings.length];
+
+        components[0] = parseComponent(componentStrings[0]);
+        String name = components[0].getName();
+        String argument = components[0].getArgument();
+
+        HashMap<String, String> flags = new HashMap<>();
+        for (int index = 1; index < componentStrings.length; index++) {
+            Component component = parseComponent(componentStrings[index]);
+            flags.put(component.getName(), component.getArgument());
+        }
+
+        switch (name) {
+            case "list": {
+                return new ListCommand();
+            }
+            case "todo": {
+                if (argument.isEmpty()) {
+                    throw new ParseException(line, "The description of a todo cannot be empty.");
+                }
+                return new CreateTodoCommand(argument);
+            }
+            case "deadline": {
+                if (argument.isEmpty()) {
+                    throw new ParseException(line, "The description of a deadline cannot be empty.");
+                }
+                DateTime by = parseDateTimeArgument(getFlagValue(line, flags, "by"));
+                return new CreateDeadlineCommand(argument, by);
+            }
+            case "event": {
+                if (argument.isEmpty()) {
+                    throw new ParseException(line, "The description of an event cannot be empty.");
+                }
+                DateTime from = parseDateTimeArgument(getFlagValue(line, flags, "from"));
+                DateTime to = parseDateTimeArgument(getFlagValue(line, flags, "to"));
+                return new CreateEventCommand(argument, from, to);
+            }
+            case "mark": {
+                if (argument.isEmpty()) {
+                    throw new ParseException(line, "A task identifier must be provided.");
+                }
+                int taskId = parseIntArgument(argument);
+                return new MarkCommand(taskId);
+            }
+            case "unmark": {
+                if (argument.isEmpty()) {
+                    throw new ParseException(line, "A task identifier must be provided.");
+                }
+                int taskId = parseIntArgument(argument);
+                return new UnmarkCommand(taskId);
+            }
+            case "delete": {
+                if (argument.isEmpty()) {
+                    throw new ParseException(line, "A task identifier must be provided.");
+                }
+                int taskId = parseIntArgument(argument);
+                return new DeleteCommand(taskId);
+            }
+            case "find": {
+                if (argument.isEmpty()) {
+                    throw new ParseException(line, "A search query must be provided.");
+                }
+                return new FindCommand(argument);
+            }
+            case "bye": {
+                return new ByeCommand();
+            }
```

Comment:
Should the indentation of `case` be reduced?

The current coding style is as such:
```
switch (condition) {
    case ABC:
        statements;
        // Fallthrough
    case DEF:
        statements;
        break;
    case XYZ:
        statements;
        break;
    default:
        statements;
        break;
}
```

Perhaps the following style is more aligned with our CS2103T Java Coding Standard Guide?
```
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```